File: totosuki/ABC/B/291.py
# import numpy as np
N = int(input())
X = list(map(int, input().split()))
X.sort()
print(sum(X[N:-N]) / (N*3))
# X = np.array(list(map(int, input().split())))
# X = np.sort(X)
# print(np.mean(np.sort(X)[N:-N]))

File: OCH_Tao/ABC/B/291.js
const input = require("fs").readFileSync("/dev/stdin", "utf8").trim().split("\n");
const N = parseInt(input[0], 10);
const X = input[1].split(" ").map(x => parseInt(x, 10));
X.sort((a, b) => a - b);
console.log(X.slice(N, 4 * N).reduce((a, b) => a + b, 0) / (3 * N));